### ISSUE TITLE
Can you cache a POST request with files

### ISSUE THREAD

**Issue by tgrandje** [requests-cache/requests-cache#841]
I have some requests to make to a french API (addresses' recognition) which use the POST method.

I'm surelly missing something but I don't think I'm caching those requests :

```
import requests_cache
from requests import Request
s = requests_cache.CachedSession(allowable_methods=('GET', 'POST'))
url_api = "https://api-adresse.data.gouv.fr/search/csv/"
files=[
    ("data", "cities_init\n02000 LAON\n59800 LILLE"),
    ("result_columns", (None, "cities_init")),
    ("result_columns", (None, "result_score")),
    ("result_columns", (None, "result_city")),
    ("result_columns", (None, "result_citycode")),
]
s.post(url_api, files=files)
print(s.cache.urls())
print(s.cache.contains(request=Request("POST", url_api, files=files)))
```

**Comment by JWCook** [comment]
Yes, caching POST requests with files is supported. You can see this is you run the request a second time, using the `from_cache` attribute:
```python
r = s.post(url_api, files=files)
print(r.from_cache)
```

However, it looks like the `contains()` method does not currently work with POSTing files. I believe this is because requests-cache needs to set a custom [form boundary](https://everything.curl.dev/http/multipart); otherwise, `requests` will randomly generate one, so subsequent request bodies will not match when looking for it in the cache.

This is something that `CachedSession.post()` does, but `CachedSession.cache.contains()` does not. Example form boundary set by requests-cache:
```
Content-Type: multipart/form-data; boundary=##requests-cache-form-boundary##
```

Example form boundary set by requests:
```
Content-Type: multipart/form-data; boundary=662cd222f12646585fe665e8dacd180f
```

I'll get that fixed.

**Comment by JWCook** [comment]
This has been fixed in v1.1.
